### PR TITLE
feat(741): fleet security-audit coverage drift audit

### PR DESCRIPTION
**Pull request by clarkemoyer**
Refs #838

## Scope — portion (a) only, and why

#838's own scope section offers the split: _"If that is too much for one PR, split: (a) the drift audit that reports which repos lack it, (b) the backfill."_ This is **(a)**. It changes nothing in any fleet repo and bumps no dependency — remediation stays with #822.

The 34-repo backfill is **not** blocked on this PR, and I left it unclaimed (scoped claim comment, no exclusive `claimed` label) so it stays in the pickup query. I could not do it myself: this session's GitHub access is an explicit repo allowlist covering only the hub plus `FFC-IN-ffcadmin.org`, `FFC-IN-FFC_Single_Page_Template`, `FFC-IN-Footer_Only_Template`, `FFC-EX-canary`. All 34 targets are outside it.

But the ordering turns out to be the right one anyway. #838 is not really "34 repos need a file" — it is "**nothing was measuring coverage**, which is why #822's scope read 13 when it was 38, then 44." A backfill fixes the 34 repos that exist today; the audit is what keeps repo 45 from being silent next month. It also satisfies the acceptance criterion _"A drift check exists that fails or reports when a Node FFC-EX repo lacks the workflow, so this cannot silently re-open."_

## What this adds

**`.github/workflows/741-fleet-security-audit-coverage.yml`** — modelled on `738` (the repo's existing fleet-file drift auditor): weekly, `GITHUB_TOKEN` only, no `environment:`, rolling issue upsert/close. It enumerates the fleet from `gh repo list` on every run rather than reading a count carried between sessions — that carried count is the specific thing #838 was opened about.

**Coverage requires both halves, and they are not interchangeable.** The audit classifies each repo into five buckets:

| Bucket | Meaning | Counts as a gap? |
| --- | --- | --- |
| `covered` | has `security-audit.yml` **and** an `audit:high` script | no |
| `partial` | exactly one half | **yes** |
| `uncovered` | Node app with neither — the #838 population | **yes** |
| `notNode` | no `package.json` (the 18 static/placeholder repos) | no — no dependency tree to audit |
| `unreadable` | fetch failed (private/rate-limit/transient) | no — coverage can't be asserted either way |

`partial` deserves the emphasis. #838 notes the two markers are perfectly correlated across the current fleet, so the bucket is empty today — but it is the state the backfill can *create*, and it is worse than `uncovered` because the repo looks wired. A workflow without the script fails on every run with `Missing script: audit:high`; a script with no workflow never runs and reports nothing. Neither produces a usable signal, and a hand sweep grepping for one marker would call both done.

`unreadable` is deliberately not a gap. Guessing "uncovered" on a failed fetch would put a repo on a remediation checklist it may not belong on — the opposite error from the one this issue is about, but the same kind.

**Schedule distribution is reported, but does not raise the issue.** #838 asks the backfill to stagger crons rather than pointing 34 more repos at `17 6 * * *`, so the report groups covered repos by cron, largest pile-up first. It is informational by design: at 44 repos across 60 minutes some sharing is unavoidable, so "no two repos share a minute" would be a rule that never goes green, and a permanently-red alert is one nobody reads. What it *does* flag unambiguously is the zero-staggering case — more than one audited repo, all on a single cron. That is the state today (verified below), and any real staggering clears it permanently. Coverage and scheduling hygiene stay separate so a red rolling issue keeps one precise meaning: **a repo with dependencies is unmonitored.**

**`scripts/fleet-audit-coverage-lib.js`** — classification, cron summary, the two parsers, and the issue body. The workflow `require`s this module (the 738 pattern), so the tested logic is the shipped logic. `hasAuditScript` parses rather than greps: a bare `"audit:high"` substring elsewhere in a `package.json` must not read as covered.

**`tests/workflow-logic/test_741_fleet_audit_coverage.py`** — 31 tests: every bucket, both `partial` directions, error-wins-over-stale-flags, the four cron cases, the parser edge cases (quoted/unquoted/absent cron, malformed JSON, non-string script value, bare mention), report rendering, and the wiring shape — read-only permissions, no environment gate, no cron collision with 738, and a guard that the workflow contains no fleet-write mechanism (`create_or_update_file` / `git push` / `create-pull-request`), since "reports only" is a claim worth enforcing.

**Docs** — safety-table row for 741 (Reads, no environment), catalog + README regenerated. The large `docs/workflow-safety-and-approvals.md` diff is Prettier reflowing the whole table because the new row widened a column — the documented behaviour in AGENTS.md, not a hand edit.

## Verification

- `tests/workflow-logic/run_all.py` — **all 24 modules pass**, including the new one (this sandbox runs the harness cleanly; no false-red).
- **Mutation-tested two headline guards**, since a guard that cannot fail is not a guard. Setting 741's cron to 738's `0 7 * * 1` fires the collision test with the expected message; relaxing `covered` from `wf && sc` to `wf || sc` fires 4 tests. Both clear on restore.
- **Parsers checked against the real canonical files**, not only fixtures — `extractCron` and `hasAuditScript` run against the actual `security-audit.yml` / `package.json` in SPT, Footer_Only_Template and FFC-EX-canary: cron `17 6 * * *`, script present, in all three.
- `check-workflow-doc-consistency.py` ✅ · `check-workflow-references.py` ✅ · `actionlint` ✅ · `prettier@3.8.1 --check` ✅ (CI-pinned version) · catalog regenerates with no drift.

**Empirical note from that check:** SPT, FOT and canary are all on `17 6 * * *`. The fleet's audit schedule is currently not staggered at all — so the first live run should report exactly that, and #838's staggering requirement is a real one, not hypothetical.

**Not verifiable pre-merge:** the first live sweep. `gh repo list`-backed enumeration needs an org-wide token, which the sandbox does not have, so the actual 44/34 split is reproduced from #838's own sweep rather than re-measured here. After merge, dispatch 741 once — the expected result is a rolling issue listing ~34 uncovered repos as a checklist, which then becomes the work-list for portion (b). A run that reports **zero** Node repos is the failure mode to watch for (it would mean the org listing or the content reads came back empty); the gather step already hard-fails on an empty fleet listing rather than reporting an empty fleet as covered.

## Gates

None — ambient-token workflow, no `environment:`, no external credentials. Normal merge queue.

---
_Generated by [Claude Code](https://claude.ai/code/session_01MabJCoScEfCm78CE6vAVic)_

### CODE REVIEW

**Comment by clarkemoyer** [comment]
**Correction to the "actionlint ✅" line in the description, and the fix for it — pushed as `3be4db1`.**

`Validate Repository` went red on the first commit, and the finding was real:

```
741-fleet-security-audit-coverage.yml:56:9: shellcheck reported issue in this script:
SC2016:info:81:13: Expressions don't expand in single quotes, use double quotes for that
```

The inline `node -e '…'` snippet used a JS template literal — `` `${lib.hasAuditScript(pkg)} …` `` — and inside a single-quoted shell argument that reads to shellcheck as a shell expansion that will never expand. Rewritten as plain concatenation rather than suppressed with a `# shellcheck disable`, so the step keeps the warning for a case where it would be a genuine bug.

**Why my pre-push check missed it:** I ran a bare `actionlint` binary. `actionlint` shells out to `shellcheck` for `run:` blocks **only if shellcheck is on PATH** — and it is not in this sandbox by default. Absent it, actionlint silently skips that entire class of check and exits 0. CI's runner image has shellcheck, so it caught what my "green" run could not see. I installed shellcheck and re-verified both directions: the pre-fix file reproduces the exact CI error locally, the fixed file is clean.

Nothing else changed — 31 unit tests, doc-consistency, reference guard and `prettier@3.8.1 --check` still pass.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: scripts/fleet-audit-coverage-lib.js [hunk 1]
```diff
@@ -0,0 +1,260 @@
...
+  }
+  const groups = [...byCron.entries()]
+    .map(([cron, repos]) => ({ cron, repos: repos.slice().sort(), count: repos.length }))
+    .sort((a, b) => b.count - a.count || a.cron.localeCompare(b.cron));
+  const distinct = groups.filter((g) => g.cron !== 'unknown').length;
+  return {
+    groups,
+    distinct,
+    staggered: !((covered || []).length >= 2 && distinct <= 1),
+  };
```
[the remark this answers is not in the document]
Comment (clarkemoyer) [reply scripts/fleet-audit-coverage-lib.js:135]: Correct, and a genuine bug — fixed in `8868bad`.

`distinct` excludes the `unknown` group by design, so a fleet of entirely dispatch-only audits gave `distinct == 0`, satisfied `distinct <= 1`, and fired the "never staggered" warning when in truth nothing was known about the schedule at all. `staggered` now gates on the count of repos whose cron is actually **known**:

```js
const knownCount = groups.filter((g) => g.cron !== 'unknown').reduce((n, g) => n + g.count, 0);
staggered: !(knownCount >= 2 && distinct <= 1)
```

Worth naming what this was, because it is the same failure this workflow exists to catch, one level down: **not-measured rendering as measured-and-bad.** The `unknown` bucket is the cron-level twin of the `unreadable` bucket, which I *did* keep out of `hasGap` for exactly this reason — I applied the principle in one place and not the other.

Four regression tests, and I mutation-tested them: restoring the old predicate fails three of them, so they genuinely hold the line.

- `test_all_unknown_crons_do_not_read_as_unstaggered`
- `test_one_known_cron_alongside_unknowns_is_not_unstaggered`
- `test_two_known_identical_crons_still_flag_despite_unknowns` — the warning must still survive unknowns in the same fleet, or one dispatch-only repo would mask a real pile-up
- `test_render_omits_stagger_warning_when_every_cron_is_unknown`

My original `test_dispatch_only_audit_copy_has_unknown_cron` used a single repo, which is below the `>= 2` threshold — so it exercised the bucket without ever reaching the predicate.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: .github/workflows/741-fleet-security-audit-coverage.yml [hunk 2]
```diff
@@ -0,0 +1,233 @@
+name: '741. Repo - Fleet Security Audit Coverage [Org]'
+run-name: 'Fleet Security Audit Coverage'
+
+# Weekly audit of whether each Node app in the FFC-EX fleet has dependency-
+# vulnerability detection wired at all. A repo counts as covered only when it
```
[the remark this answers is not in the document]
Comment (clarkemoyer) [reply .github/workflows/741-fleet-security-audit-coverage.yml]: Right — fixed in `8868bad`, along with the catalog entry (the other thread is the same defect seen from the generated side).

The header wrapped `dependency-vulnerability` across lines, and the catalog generator joins header comment lines with a space, so it shipped as `dependency- vulnerability` to `docs/workflow-catalog.json` and from there to the public rendering at ffcadmin.org/automation/. Rewrapped so the word stays whole; catalog regenerated and confirmed clean.

I first added an inline note in the header explaining the hazard — which promptly shipped *itself* into the public catalog description, quoting the typo verbatim. Removed. Maintenance guidance does not belong in a field that is rendered publicly, so it is now a test instead: `test_741_header_never_wraps_mid_hyphenated_word` asserts both the mechanical cause (no header line ends in a hyphen) and the outcome (no `\w+- \w+` in 741's generated description).

**One thing to flag rather than fix here:** `727`'s description carries the identical pattern — `squash- or` — and predates this PR, so I scoped the guard to 741 rather than let an unrelated pre-existing typo fail this branch. Widening it to a repo-wide check in `check-workflow-doc-consistency.py`, plus the one-line rewrap in 727, is a clean follow-up for whoever picks it up.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: docs/workflow-catalog.json [hunk 3]
```diff
@@ -1847,6 +1847,25 @@
...
...n the FFC-EX fleet has dependency- vulnerability detection wired at all. A repo counts as covered only when it has BOTH `.github/workflows/security-audit.yml` and an `audit:high` script in package.json \u2014 the workflow without the script fails on every run, the script without the workflow never runs. Findings are upserted into a single rolling issue that auto-closes once the fleet is fully covered.  Why (#838): 34 of the 44 Node FFC-EX repos had no detection of any kind, so a published advisory against a charity site was invisible until somebody swept by hand \u2014 and nothing was measuring that, which is why #822's scope was estimated at 13 repos when the real number was 38. This workflow re-derives the fleet from the org listing every run rather than trusting a count carried between sessions.  Read-only: GITHUB_TOKEN reads public repo contents org-wide and lists the org's public repos; the only write is the rolling issue in THIS repo. No external API, no environment gate. It reports coverage only and never bumps a dependency \u2014 remediation belongs to #822. Classification/report logic lives in scripts/fleet-audit-coverage-lib.js (unit-tested under tests/workflow-logic/).",
```
[the remark this answers is not in the document]
Comment (clarkemoyer) [reply docs/workflow-catalog.json]: Same defect as the workflow-header thread, seen from the generated side — fixed at the source in `8868bad` and the catalog regenerated, so `741`'s description now reads `dependency-vulnerability`. Full detail (including why the guard is scoped to 741 rather than repo-wide, given `727`'s pre-existing `squash- or`) is in the reply on [that thread](https://github.com/FreeForCharity/FFC-Cloudflare-Automation/pull/840#discussion_r3649640842).

---
_Generated by [Claude Code](https://claude.ai/code)_
